### DISCUSSION TITLE
Are variadic parameters preferred for attributes?

### DISCUSSION THREAD

**Discussion by andreasley** [elementary-swift/elementary#35] [General]
I'd intuitively want to write this:

```swift
img(src: "test.jpg", width: "300", height: "300")
```

But Elementary expects this:

```swift
img(.src("test.jpg"), .custom(name: "width", value: "300"), .custom(name: "height", value: "300"))
```

Would you consider having convenience initializers like the following one added to the library?

```swift
extension img
{
    init(src: String, width: String? = nil, height: String? = nil) {
        var attributes = [HTMLAttribute<HTMLTag.img>]()
        attributes.append(.src(src))
        if let width { attributes.append(.custom(name: "width", value: width)) }
        if let height { attributes.append(.custom(name: "height", value: height)) }
        self.init(attributes: attributes)
    }
}
```

Also, would additional `HTMLAttribute` be welcome (e.g. `.width` and `height`)?

**Comment by sliemeobn** [comment]
_> Would you consider having convenience initializers like the following one added to the library?_

I stayed clear of "convenience initializers", because HTML is just such an unwieldy beast. I feel like whatever we'd choose as "normal" attributes, you'll quickly find another use case. And then you need a ton of overloads for each element (eg: to combine it with other attributes...).

Also, given how easy it is to just bring-you-own-initializers in your package, I feel this is not really a good step for now.

_> Also, would additional HTMLAttribute be welcome (e.g. .width and height)?_

Absolutely! Those are just simply missing, and frankly I am a bit surprised that these are not there yet :/
If you want to give adding `.width` and `.height` a go, happy to merge a PR!
Otherwise this would go high on my list as well.

**Reply by sliemeobn** [reply]
width and height are now added
#42

**Reply by andreasley** [reply]
Thank you! I wanted to check the whole list of available attributes but didn't get around to it yet. I will eventually. :)

**Reply by sliemeobn** [reply]
"the whole list" is quite overwhelming to be honest, and many, many are quite esoteric or widely unused. so I am not sure going for completeness is a real goal here...

**Reply by andreasley** [reply]
I totally agree, but I think there are still a few essential ones missing. Maybe 10 or so.
